src/vulkan/device.cc

```diff
@@ -20,8 +20,24 @@
 #include "src/make_unique.h"
 
 namespace amber {
-
 namespace vulkan {
+namespace {
+
+bool AreAllRequiredFeaturesSupported(
+    const VkPhysicalDeviceFeatures& features,
+    const VkPhysicalDeviceFeatures& required_features) {
+  const VkBool32* supported = reinterpret_cast<const VkBool32*>(&features);
+  const VkBool32* required =
+      reinterpret_cast<const VkBool32*>(&required_features);
+  for (size_t i = 0; i < (sizeof(VkPhysicalDeviceFeatures) / sizeof(VkBool32));
```

Comment:
This is scary. Is this how vulkan normally checks features? Casting the VkPhysicalDeviceFeatures to a VkBool32 seems .... wrong?

Why not just pass the std::vector<Feature> into here, then loop over that vector, have a switch(feature) and check each feature? It would be a lot clearer what's going on.

The switch below basically moves up here?

Reply:
Maybe not even a switch, simple if statements might be simplest:

if (feature == kFeatureFoo && features.feature_foo == Vk_FALSE)
  return false;
if (feature == kFeatureBar && features.feature_bar == VK_FALSE)
  return false;